David-Lyu/git-self-host
# Docker dev env

Integrated docker with our Server Side

WIP need to add git into the server

New functionality:
- ignored: 
  - /server/tmp/
  - /server/model/database/sqlite-database.db

Test to reproduce:
- To test production and init.sh run

```

docker compose --profile production up

```
- To test development and air
```

docker compose --profile dev up
```

- You can try to see it "Hot reload" doesn't really hot reload but remakes binary automatically. Didn't read too much on code but read a reddit saying it was good. Haha

- Init.sh should:
  - [ ] check OS
  - [ ] check Linux Distro
  - [ ] check if its from docker
  - [ ] install git
  - [ ] run binary
